CATS/Problem/Parser.pm

```diff
@@ -220,6 +244,12 @@ sub validate
 
     $check_order->($problem->{samples}, 'sample');
 
+    my @samples = values %{$problem->{samples}};
```

Comment:
Except for the fact that it is more readable than: `for (sort {$a->{rank} <=> $b->{rank}} values %{$problem->{samples}}) {`